editor/debugger/debug_adapter/debug_adapter_protocol.cpp

```diff
@@ -661,11 +674,15 @@ int DebugAdapterProtocol::parse_variant(const Variant &p_var) {
 
 void DebugAdapterProtocol::parse_object(SceneDebuggerObject &p_obj) {
 	// If the object is not on the pending list, we weren't expecting it. Ignore it.
-	ObjectID object_id = p_obj.id;
-	if (!object_pending_set.erase(object_id)) {
+	ObjectID remote_object_id = p_obj.id;
+	if (!object_pending_list.has(remote_object_id)) {
 		return;
 	}
 
+	ERR_FAIL_COND(!object_list.has(remote_object_id));
+	const Ref<ThreadData> &thread = object_pending_list[remote_object_id];
```

Comment:
```suggestion
	const Ref<ThreadData> thread = object_pending_list[remote_object_id];
```
Since it is erased after the reference isn't safe I'd say